lib/CHeroHandler.cpp

```diff
@@ -117,9 +118,15 @@ CHeroClass * CHeroClassHandler::loadFromJson(const JsonNode & node, const std::s
 		heroClass->primarySkillHighLevel.push_back(node["highLevelChance"][pSkill].Float());
 	}
 
-	for(const std::string & secSkill : NSecondarySkill::names)
+	for(auto skillPair : node["secondarySkills"].Struct())
 	{
-		heroClass->secSkillProbability.push_back(node["secondarySkills"][secSkill].Float());
+		int probability = skillPair.second.Integer();
+		VLC->modh->identifiers.requestIdentifier(skillPair.second.meta, "skill", skillPair.first, [heroClass, probability](si32 skillID)
+		{
+			if(heroClass->secSkillProbability.size() <= skillID)
+				heroClass->secSkillProbability.resize(skillID + 1, -1); // -1 = override with default later
+			heroClass->secSkillProbability[skillID] = probability;

```

Comment:
Far out of scope of this PR, but seems like heroclass has problems with `int` in persistent data. And `probability` here should be `si32`